cmd/detectExecuteScan.go

```diff
@@ -253,9 +254,24 @@ func exitCodeMapping(exitCodeKey int) string {
 
 func getDetectScript(config detectExecuteScanOptions, utils detectUtils) error {
 	if config.ScanOnChanges {
+		log.Entry().Infof("Using Detect Rescan script")
 		return utils.DownloadFile("https://raw.githubusercontent.com/blackducksoftware/detect_rescan/master/detect_rescan.sh", "detect.sh", nil, nil)
 	}
-	return utils.DownloadFile("https://detect.synopsys.com/detect.sh", "detect.sh", nil, nil)
+	if piperutils.ContainsStringPart(config.CustomEnvironmentVariables, "DETECT_LATEST_RELEASE_VERSION") {
+		releaseVersion := ""
+		for _, i := range config.CustomEnvironmentVariables {
```

Comment:
Does it check "only" the config.CustomEnvironmentVariables, which are e.g. set in the internal piper-defaults.yml or does it also check exported env variables e.g.
`export DETECT_LATEST_RELEASE_VERSION=123
./piper detectExecuteScan ...
`
?

Reply:
Yes, you are right. Only the config.CustomEnvironmentVariables are checked with this method. I will see if I can add the check for environment variables in general.

Reply:
If this behaviour is desired then there should be a test case for an environment variable too in addition to the config.CustomEnvironmentVariable check

Reply:
Check for the variable set from OS Environment and the corresponding test has been added. Thanks for your feedback!